syft/source/all_layers_resolver.go

```diff
@@ -120,18 +120,28 @@ func (r *allLayersResolver) FilesByPath(paths ...string) ([]Location, error) {
 }
 
 // FilesByGlob returns all file.References that match the given path glob pattern from any layer in the image.
+//
+//nolint:gocognit
 func (r *allLayersResolver) FilesByGlob(patterns ...string) ([]Location, error) {
 	uniqueFileIDs := file.NewFileReferenceSet()
 	uniqueLocations := make([]Location, 0)
 
 	for _, pattern := range patterns {
 		for idx, layerIdx := range r.layers {
-			results, err := r.img.Layers[layerIdx].Tree.FilesByGlob(pattern, filetree.FollowBasenameLinks, filetree.DoNotFollowDeadBasenameLinks)
+			results, err := r.img.Layers[layerIdx].SquashedTree.FilesByGlob(pattern, filetree.FollowBasenameLinks, filetree.DoNotFollowDeadBasenameLinks)
 			if err != nil {
 				return nil, fmt.Errorf("failed to resolve files by glob (%s): %w", pattern, err)
 			}
 
 			for _, result := range results {
+				found, _, err := r.img.Layers[layerIdx].Tree.File(result.RealPath)
+				if err != nil {
+					return nil, fmt.Errorf("failed to resolve file during glob resolution %s: %w", result.RealPath, err)
```

Comment:
question: should we warn, return an error, or continue without note here?

I'm curious about what the error conditions are in `File()` and what that might hint at.

Reply:
I'll dig in to this, but seeing as there's a `bool` return, I maybe incorrectly assumed a "not found" should just be that...

Reply:
It turns out one cannot look in the layer to determine if a file exists, it just needs to use the `SquashedTree`, so removed this bit anyway.